credentials.go

```diff
@@ -153,6 +158,25 @@ Error: %w`, err)
 	return provider, creds.Source, err
 }
 
+func webIdentityCredentialsProvider(ctx context.Context, awsConfig aws.Config, c *Config) (aws.CredentialsProvider, error) {
+	ar := c.AssumeRole
```

Comment:
I reused `config.AssumeRole` 'block' here, although not every option is valid for 'assumeRoleWithWebIdentity'.
@gdavison - let me know if it's ok, or if you prefer to have something like `config.WebIdentityAssumeRole`

In https://github.com/hashicorp/terraform-provider-aws/issues/22907#issuecomment-1048371318 you accepted combining both at the provider level, but I don't know how it should be done here.

Reply:
I'm reconsidering that statement 🙂 Since the two operations support different options, it would probably be better to explicitly choose one or the other, both here and in the provider.

Reply:
ok, I changed the Config and created a separate struct for AssumeRoleWithWebIdentity